tests/playwright/pages/library-kits-page.js

```diff
@@ -0,0 +1,253 @@
+    }
+
+    async checkTheAttorneyTagIsUnchecked() {
+        await expect( this.attorneyTag, `The attorney tag checkbox is still checked` ).not.toBeChecked();
+    }
+
+    async checkAllKitsAreNowPresent() {
+        await expect( await this.kitTitles.count(), `All the kits are not back into view after filter is cleared` ).toBeGreaterThan( 50 );
+    }
+
+    async searchForKit( kit ) {
+        await this.mainKitSearchEntryBox.type( Object.keys( kit )[ 0 ], { delay: 50 } );
+        await this.page.waitForTimeout( 500 );
+    }
+
+    async selectViewKitDemo() {
+        await this.viewKitDemo.click();
+        await this.page.waitForLoadState( 'networkidle' );
+        await this.page.waitForTimeout( 500 );
+    }
+
+    async checkKitDemoHeading( kit ) {
+        await expect( this.kitH1 ).toContainText( kit );
+    }
+
+    async checkSearchedKitIsPresent( kit ) {
+        const kitslist = await this.kitTitles.allTextContents();
+        await expect( kitslist.includes( Object.keys( kit )[ 0 ] ) ).toBeTruthy();
+    }
+
+    async checkFreePlan() {
+        await this.page.check( this.freePlanSelector );
+    }
+
+    async checkProPlan() {
+        await this.page.check( this.proPlanSelector );
+    }
+
+    async checkExpertPlan() {
+        await this.page.check( this.expertPlanSelector );
+    }
+
+    async goToFavoritesPage() {
+        await this.favoritesPage.click();
+        await this.page.waitForLoadState( 'networkidle' );
+    }
+
+    async goToAllWebsiteKits() {
+        await this.allWebsiteKits.click();
+    }
+
+    async selectSortoption( selectOption ) {
+        await this.sortOptionsDropDown.selectOption( { value: selectOption } );
+    }
+
+    async checkBlogTag() {
+        await this.page.check( this.blogTagSelector );
+    }
+
+    async selectClearAllFilters() {
+        await this.page.locator( '.e-kit-library__filter-indication-button span' ).click();
+    }
+
+    async selectBackToLibraryButton() {
+        await this.page.locator( '.e-kit-library__header-back' ).click();
+    }
+
+    async selectOverviewButton() {
+        await this.page.locator( '#eps-app-header-btn-overview' ).click();
+    }
+
+    async validateKitDescription( descriptionText ) {
+        await expect( this.page.locator( '.e-kit-library__item-sidebar-description' ) ).toContainText( descriptionText );
+    }
+
+    async checkKitPageHeading( headingText ) {
+        await expect( this.kitH1 ).toContainText( headingText );
+    }
+
+    async addKitToFavorites( kitName ) {
+        await this.page.waitForLoadState( 'networkidle' );
+        await this.goToAllWebsiteKits();
+        await this.page.waitForSelector( this.kitTitlesSelector );
+        const kitTitlesList = await this.kitTitles.allTextContents();
+        const kitIndex = kitTitlesList.indexOf( kitName );
+        await this.facoriteIcon.nth( kitIndex ).click();
+        await this.page.waitForLoadState( 'networkidle' );
+        await this.page.waitForTimeout( 1000 );
```

Comment:
When we go to the favorites page, there are no backend calls that are made, and also there is no specific selector to check for to know 100% that everything is loaded. But we need EVERYTHING to load because we are doing a .allTextContents to grab all the names. We tried with Niv to get to avoid it. Couldn't figure it out. Maybe in the future, we can update it if we get a solution.